workflows/prognostic_run_diags/fv3net/diagnostics/prognostic_run/views/matplotlib.py

```diff
@@ -89,7 +102,71 @@ def plot_2d_matplotlib(
         template.render(
             image_data=data,
             runs=sorted(run_diags.runs),
-            variables_to_plot=variables_to_plot,
+            variables_to_plot=sorted(variables_to_plot),
+            varfilter=varfilter,
+        )
+    )
+
+
+def plot_cube_matplotlib(
+    run_diags: RunDiagnostics,
+    run_metrics: pd.DataFrame,
+    varfilter: str,
+    metric_names: Mapping[str, str] = None,
+    **opts,
+) -> str:
+    """Plot horizontal maps of cubed-sphere data for all diagnostics whose name includes
+    varfilter. All matching diagnostics must have tile, x and y dimensions
+    and each dataset in run_diags must include lat/lon/latb/lonb coordinates."""
+
+    data = defaultdict(dict)
+
```

Comment:
This function does a few things that could be the responsibility of `RunDiagnostics` or some routine farther up the call-stack. It has
- grid related logic
- Querying logic for metrics and metric names.
- varfilter for selecting the data to plot. Since this is such a common pattern, I would add a method which selects the data matching a varfilter to `RunDiagnostics`. I wouldn't have asked for this refactor if it weren't for the pile on effect.

I would move these pure-data operations to the calling routine, and pass this routine data which is ready for visualization.

Reply:
Okay, I added a `RunMetrics` dataclass and added a couple new methods to `RunDiagnostics` which simplifies this function. It's now similar complexity to the existing `plot_2d_matplotlib` function.

Reply:
`RunMetrics` looks great!

Reply:
Thanks